vnext/ReactWindowsCore/WinRTWebSocketResource.h

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// clang-format off
+
+#pragma once
+
+//#include <Mso/eventWaitHandle/eventWaitHandle.h>
+#include <winrt/Windows.Networking.Sockets.h>
+#include <winrt/Windows.Storage.Streams.h>
+#include <IWebSocketResource.h>
+
+#include <future>
+#include <queue>
+
+namespace Microsoft::React
+{
+
+class WinRTWebSocketResource : public IWebSocketResource
+{
+  winrt::Windows::Foundation::Uri m_uri;
+  winrt::Windows::Networking::Sockets::MessageWebSocket m_socket;
+  winrt::Windows::Networking::Sockets::MessageWebSocket::MessageReceived_revoker m_revoker;
+  winrt::Windows::Storage::Streams::DataWriter m_writer;
+
+  std::atomic_bool m_connectRequested;
+  std::atomic_bool m_sendRequested;
+  // Mso::ManualResetEvent m_connectPerformed;
+  // Mso::ManualResetEvent m_sendPerformed;
+  std::promise<void> m_connectPerformed;
+
+  CloseCode m_closeCode;
+  std::string m_closeReason;
+  std::queue<std::pair<std::string, bool>> m_writeQueue;
+
+  std::function<void()> m_connectHandler;
+  std::function<void()> m_pingHandler;
+  std::function<void(std::size_t)> m_writeHandler;
+  std::function<void(std::size_t, const std::string&)> m_readHandler;
```

Comment:
Why do we need a length param instead of relying on the string?  (Also, would be great if we could document these).

Reply:
This was influenced by the Beast implementation's callback mechanism.
Keeping for interface compatibility, for now.